rpcclient/infrastructure.go

```diff
@@ -1113,6 +1127,25 @@ type ConnConfig struct {
 	EnableBCInfoHacks bool
 }
 
+// getAuth returns the username and passphrase that will actually be used for
+// this connection.  This will be the result of checking the cookie if a cookie
+// path is configured; if not, it will be the user-configured username and
+// passphrase.
+func (config *ConnConfig) getAuth() (username, passphrase string, err error) {
```

Comment:
Bitcoin Core falls back to cookie authentication if a password is not found, we might want to replicate that functionality rather than fall back to username/password authentication if a cookie is not found.

Additionally, there is no example of cookie creation - this is a relevant feature, and it is currently undefined.

Reply:
> Additionally, there is no example of cookie creation - this is a relevant feature, and it is currently undefined.

Hmm, what functionality are you looking for exactly?  Making the btcd daemon support creating a cookie file?

Reply:
Yup - something like in https://github.com/bitcoin/bitcoin/blob/master/src/rpc/request.cpp
Just so users don't have to generate using core first, and this can be a little more feature complete.

Reply:
Out of scope for this PR though